src/libAtomVM/bitstring.c

```diff
@@ -82,10 +82,31 @@ bool bitstring_insert_any_integer(uint8_t *dst, avm_int_t offset, avm_int64_t va
     return true;
 }
 
+static bool invalid_codepoint(int32_t v)
+{
+    return (v < 0) || (0x10FFFF < v) || (0xD800 <= v && v <= 0xDFFF);
```

Comment:
You might not like the way I have written the inequalities.  Happy to change them to be more conventional.

Reply:
they right indeed, but it takes me always a little more time to read them, so I rather go with:
`return (v < 0) || ( v > 0x10FFFF) || (v >= 0xD800 && v <= 0xDFFF);`